### PR TITLE
Add terraform validator

### PR DESCRIPTION
**Pull request by manitgupta**
- Adds a Terraform Validator for Terraform samples being added to the repo. It verifies the sample structure and the `plan` generated for it via `terraform validate`.
- Formats the existing bulk migration example to comply with Terraform validator using `terraform fmt`.

### CODE REVIEW

File: .github/workflows/spanner-terraform-validator.yml [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+          echo "Verifying terraform configuration..."
+          terraform -chdir="$SAMPLE" init
+          terraform -chdir="$SAMPLE" validate
+        
+          echo "Verifying sample structure..."
+          for FILE in $FILE_LIST; do
+            if [[ ! -f "$SAMPLE/$FILE" ]]; then
+              echo "ERROR: Missing file: $SAMPLE/$FILE"
+              FAILED_SAMPLES+=("$SAMPLE")
+              break
```
Comment (asthamohta) [inline .github/workflows/spanner-terraform-validator.yml:52]: What if multiple files are missing? Should we print all missing files?

Comment (manitgupta) [reply .github/workflows/spanner-terraform-validator.yml:52]: It was just being lazy 😛
Okay will collect the list and print it!

Comment (manitgupta) [reply .github/workflows/spanner-terraform-validator.yml]: Done!

**Review by asthamohta — APPROVED** [review]

**Comment by bharadwaj-aditya** [comment]
Can you add some comments/ docs with use cases on the create sample TF scipt?

**Comment by manitgupta** [comment]
> Can you add some comments/ docs with use cases on the create sample TF scipt?

As in comments/docs on the samples that will follow?

**Review by bharadwaj-aditya — APPROVED** [review]
LGTM

**Review by damondouglas — CHANGES_REQUESTED** [review]
How does this work with the existing method to autogenerate terraform based on the template spec? If the existing solution isn't compatible with what already exists in the rest of the repository is fine but would be helpful to know either way.

**Comment by manitgupta** [comment]
This is independent of that and is only localised to Spanner specific templates.

The existing auto-generated template only focuses on Dataflow while the samples we will add will focus on other resources relevant to an IaaC configuration with respect to the Spanner templates.

**Review by damondouglas — APPROVED** [review]

**Pull request merged**